parquet-hadoop/src/main/java/org/apache/parquet/filter2/compat/RowGroupFilter.java

```diff
@@ -40,15 +45,34 @@
 public class RowGroupFilter implements Visitor<List<BlockMetaData>> {
   private final List<BlockMetaData> blocks;
   private final MessageType schema;
+  private final List<FilterLevel> levels;
+  private final Supplier<FSDataInputStream> streamSupplier;
+
+  public enum FilterLevel {
+    STATISTICS,
+    DICTIONARY
+  }
 
   public static List<BlockMetaData> filterRowGroups(Filter filter, List<BlockMetaData> blocks, MessageType schema) {
     checkNotNull(filter, "filter");
     return filter.accept(new RowGroupFilter(blocks, schema));
   }
 
+  public static List<BlockMetaData> filterRowGroups(List<FilterLevel> levels, Filter filter, List<BlockMetaData> blocks, MessageType schema, Supplier<FSDataInputStream> streamSupplier) {
```

Comment:
I think I understand what you're saying here.  The only important thing to maintain is that we don't want to prematurely load data that we're not going to process.  The most recent version of the dictionary read JIRA has something similar to what you're describing, but it requires that the row group be loaded in order to read the dictionary.  This isn't effective because we want to read the dictionary prior to loading the data.